expat/lib/xmlparse.c

```diff
@@ -6295,6 +6295,30 @@ poolStoreString(STRING_POOL *pool, const ENCODING *enc,
   return pool->start;
 }
 
+static size_t
+poolBytesToAllocateFor(int blockSize)
+{
+  /* Unprotected math would be:
+  ** return offsetof(BLOCK, s) + blockSize * sizeof(XML_Char);
+  **
+  ** Detect overflow, avoiding _signed_ overflow undefined behavior
+  ** For a + b * c we check b * c in isolation first, so that addition of a
+  ** on top has no chance of making us accept a small non-negative number
+  */
+  const int stretchedBlockSize = (int)((unsigned)blockSize * sizeof(XML_Char));
```

Comment:
Ouch, good point!

`XML_Char` can be `wchar_t` which would be 4 bytes on Linux afair.